plugins/woocommerce/src/Blocks/Shipping/ShippingController.php

```diff
@@ -488,4 +492,29 @@ function( $location ) {
 
 		return $served;
 	}
+
+	/**
+	 * Check if legacy local pickup is activated in any of the shipping zones or in the Rest of the World zone.
+	 *
+	 * @since 8.8.0
+	 *
+	 * @return bool
+	 */
+	public static function is_legacy_local_pickup_active() {
+		$shipping_zones      = WC_Shipping_Zones::get_zones();
+		$shipping_zones['0'] = array(
+			'zone_name'        => __( 'Rest of the World', 'woocommerce' ),
+			'shipping_methods' => WC_Shipping_Zones::get_zone( 0 )->get_shipping_methods(),
+		);
```

Comment:
Out of curiosity, how come get_zones() doesn't return the "Rest of the world"?

Can we do this instead how it is already being done in [other places](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-shipping-zones-v2-controller.php#L113)?
This way, it seems like we're just overwriting `id 0`'s `zone_name`.

```suggestion
		$rest_of_the_world = WC_Shipping_Zones::get_zone_by( 'zone_id', 0 );
		$shipping_zones     = WC_Shipping_Zones::get_zones();
		array_unshift( $shipping_zones, $rest_of_the_world->get_data() );
```

Reply:
> Out of curiosity, how come get_zones() doesn't return the "Rest of the world"?

All shipping zones, except "Rest of the world", are stored in the database table `{$wpdb->prefix}woocommerce_shipping_zones`. This is the corresponding section in the codebase: https://github.com/woocommerce/woocommerce/blob/72f5db9b6d2b1075284965caaa71fad7d15b3b4b/plugins/woocommerce/includes/data-stores/class-wc-shipping-zone-data-store.php#L303-L306

> Can we do this instead, how it is already being done in [other places](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-shipping-zones-v2-controller.php#L113)?
This way, it seems like we're just overwriting `id 0z's `zone_name`.

Good catch, @wavvves. For testing purposes, I've adjusted this locally instead of committing your suggestion.